implementations/rust/ockam/ockam_api/src/nodes/service/secure_channel.rs

```diff
@@ -90,8 +92,18 @@ impl NodeManager {
         authorized_identifiers: Option<Vec<IdentityIdentifier>>,
         credential_exchange_mode: CredentialExchangeMode,
         timeout: Option<Duration>,
+        identity_ctx: Option<(CowStr<'_>, &Context)>,
     ) -> Result<Address> {
-        let identity = self.identity()?.async_try_clone().await?;
+        let identity = if let Some((identity, ctx)) = identity_ctx {
+            CliState::new()?
+                .identities
+                .get(&identity)?
+                .config
+                .get(ctx, self.vault()?)
```

Comment:
Just to make sure we can handle more situations, let's do the following.
- Try building the identity with the node's vault, as you have right now, but don't bubble up the error if it fails.
- If it fails, try again with the default vault as a fallback, and if that fails again, then we bubble up the error.